integrations/pooltogether-v4/components/form-yield-source-prize-pool-deposit.tsx

```diff
@@ -0,0 +1,158 @@
+import { useEffect, useState } from 'react'
+
+import * as Form from '@radix-ui/react-form'
+import { useErc20Approve, useErc20Decimals } from '@turbo-eth/erc20-wagmi'
+import { BigNumber, ethers } from 'ethers'
+import { useDebounce } from 'usehooks-ts'
+import { useAccount, useWaitForTransaction } from 'wagmi'
+
+import { useLoadContractFromChainId } from '@/actions/pooltogether-v4/hooks/use-load-contract-from-chain-id'
+import { useUsdcApproval } from '@/actions/pooltogether-v4/hooks/use-usdc-approval'
+import { useUserBalanceDeposit } from '@/actions/pooltogether-v4/hooks/use-user-balance-deposit'
+import { usePoolTogetherPrizePoolDepositTo } from '@/actions/pooltogether-v4/pooltogether-v4-wagmi'
+import { PRIZE_POOL_CONTRACT } from '@/actions/pooltogether-v4/utils/prize-pool-contract-list'
+import { USDC_CONTRACT } from '@/actions/pooltogether-v4/utils/usdc-contract-list'
+import { Checkbox } from '@/components/ui/checkbox'
+
+export function FormDeposit() {
+  const [isChecked, setIsChecked] = useState(false)
+  const [approvalAmount, setApprovalAmount] = useState<BigNumber>(BigNumber.from(0))
+  const [submitDeposit, setSubmitDeposit] = useState<boolean>(false)
+  const [isValidAmount, setValidAmount] = useState<boolean>(true)
+
+  const { address } = useAccount()
+  const prizePoolAddress = useLoadContractFromChainId(PRIZE_POOL_CONTRACT)
+  const usdcAddress = useLoadContractFromChainId(USDC_CONTRACT)
+
+  const { data: decimals } = useErc20Decimals({ address: usdcAddress })
+  const POWER: any = decimals != undefined ? BigNumber.from(10).pow(decimals) : BigNumber.from(10).pow(6)
+
+  const userBalance = useUserBalanceDeposit()
+  const isApproved = useUsdcApproval(userBalance)
+
+  const [depositAmount, setDepositAmount] = useState('')
+  const debouncedDepositAmount = useDebounce((POWER as any) * Number(depositAmount), 500)
+
+  // @ts-ignore
+  const { data, write: depositToken } = usePoolTogetherPrizePoolDepositTo({
+    address: prizePoolAddress,
+    args: [address, debouncedDepositAmount],
+    enabled: isApproved && Boolean(debouncedDepositAmount),
+    overrides: {
+      gasLimit: BigNumber.from(750000),
+    },
+  })
```

Comment:
Let's try not use @ts-ignore, specially when dealing with the contract hooks, if there's a typescript error is because some of the hooks input is not well formatted
```suggestion
  const { data, write: depositToken } = usePoolTogetherPrizePoolDepositTo({
    mode: 'recklesslyUnprepared',
    address: prizePoolAddress,
    args: [address || '0x0', BigNumber.from(debouncedDepositAmount)],
    overrides: {
      gasLimit: BigNumber.from(750000),
    },
  })
```